### PR TITLE
Implement Kick Command

### PR DESCRIPTION
**Pull request by aeltoum**
[Trello Ticket](https://trello.com/c/0w8HFgug)

- Added Kick command functionality alongside proper error handling and reason logging
- Updated Mute command to properly handle errors and log mute reasons

_Note: This ticket required adding the variable muteReason to the constructor of the User class in order to store reasons properly_

### CODE REVIEW

File: Backend/Models/Messages/CommandMessage.cs [hunk 1]
```diff
@@ -122,10 +144,55 @@ private void ProcessMute()
...
             }
+            
+            if (userToMute.IsMuted == true)
+            {
+                CLogger.Error("Command not invoked: User is already muted");
+                SendRefuse("User is already muted");
+                return;
+            }
 
+            userToMute.MuteReason = _with;
```
Comment (OmarQurashi868) [inline Backend/Models/Messages/CommandMessage.cs]: We need to update the user in the DB here using userToMute.UpdateInDB()

File: Backend/Models/Users/UserManager.cs [hunk 2]
```diff
@@ -14,7 +14,10 @@ public static void Mute(User user)
             user.UpdateToDB();
         }
         public static void Unmute(User user) { }
-        public static void Kick(User user) { }
+        public static void Kick(User user) 
+        {
+            UserManager.Disconnect(user);
+        }
```
Comment (OmarQurashi868) [inline Backend/Models/Users/UserManager.cs:23]: Separate this function from its neighbors with new lines

File: Backend/Models/Users/User.cs [hunk 3]
```diff
@@ -54,6 +54,19 @@ public bool IsBanned
             get { return _isBanned; }
             set { _isBanned = value; }
         }
+        private string _muteReason = "";
```
Comment (OmarQurashi868) [inline Backend/Models/Users/User.cs]: New line here

File: Backend/Models/Messages/CommandMessage.cs [hunk 4]
```diff
@@ -122,10 +144,54 @@ private void ProcessMute()
...
+            {
+                CLogger.Error("Command not invoked: User is already muted");
+                SendRefuse("User is already muted");
+                return;
+            }
+
+            UserManager.Mute(userToMute, _with);
+            SendAccept();
+            CLogger.Event($"User Muted: {_target}. Reason: {_with}");
+            SendAlert($"User Muted: {_target}. Reason: {_with}");
```
Comment (OmarQurashi868) [inline Backend/Models/Messages/CommandMessage.cs]: This is better as "User has been muted"

File: Backend/Models/Messages/CommandMessage.cs [hunk 5]
```diff
@@ -122,10 +144,54 @@ private void ProcessMute()
                 SendRefuse("User not found");
                 return;
             }
+            
+            if (userToMute.IsMuted == true)
```
Comment (ahmed-okkio) [inline Backend/Models/Messages/CommandMessage.cs]: you can get rid of the == true here it'll still work

File: Backend/Models/Messages/CommandMessage.cs [hunk 6]
```diff
@@ -122,10 +144,54 @@ private void ProcessMute()
...
+
+        private void ProcessKick()
+        {
+            if (_sender == null)
+            {
+                CLogger.Error("Command not invoked: Missing sender");
+                return;
+            }
+
+            if (UserManager.IsUserAdmin(_sender) == false)
```
Comment (ahmed-okkio) [inline Backend/Models/Messages/CommandMessage.cs:169]: == false can go for a ! at the start

**Comment by ahmed-okkio** [comment]
looks good otherwise 👍

**Review by OmarQurashi868 — APPROVED** [review]

**Pull request merged**